analyzer/pkg/dpkg/copyright.go

```diff
@@ -0,0 +1,90 @@
+package dpkg
+
+import (
+	"bufio"
+	"bytes"
+	"github.com/aquasecurity/fanal/analyzer"
+	"github.com/aquasecurity/fanal/types"
+	"github.com/aquasecurity/fanal/utils"
+	dio "github.com/aquasecurity/go-dep-parser/pkg/io"
+	classifier "github.com/google/licenseclassifier/v2/assets"
+	"io"
+	"regexp"
+	"strings"
+)
+
+const LicenseAdder = "dpkg-license-adder"
+
+var (
+	cl, _                = classifier.DefaultClassifier()
+	copyrightFileRegexp  = regexp.MustCompile(`^usr/share/doc/([0-9A-Za-z_.-]+)/copyright$`)
+	commonLicensesRegexp = regexp.MustCompile(`/?usr/share/common-licenses/([0-9A-Za-z_.+-]+[0-9A-Za-z+])`)
+)
+
+type License struct {
+	Pkg      string
+	Licenses string
+}
+
+// parseCopyrightFile parses /usr/share/doc/*/copyright files
+func parseCopyrightFile(content dio.ReadSeekerAt, filePath string) (*analyzer.AnalysisResult, error) {
+	var licenses []string
+	var buf bytes.Buffer
+
+	tee := io.TeeReader(content, &buf) // Save stream in buffer for re-read with 'licenseclassifier'
+	scanner := bufio.NewScanner(tee)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// "License: *" pattern is used
+		if strings.HasPrefix(line, "License:") {
+			l := strings.TrimSpace(line[8:])
+			if !utils.StringInSlice(l, licenses) {
```

Comment:
As Alim suggested let's  use build in Go 18 method https://pkg.go.dev/golang.org/x/exp/slices#Contains

Reply:
changed

Reply:
changed